File: spring-grpc-client/app/src/main/java/com/example/springgrpc/app/GrpcClientConfiguration.java
package com.example.springgrpc.app;

import com.example.springgrpc.client.HelloGrpcClient;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class GrpcClientConfiguration {
    @Bean(destroyMethod = "close")
    public HelloGrpcClient helloGrpcClient(GrpcTargetProperties properties) {
        return HelloGrpcClient.connect(
            properties.getHost(),
            properties.getPort(),
            properties.getTimeoutSeconds()
        );
    }
}

File: spring-grpc-client-kotlin/app/src/main/kotlin/com/example/springgrpc/app/GrpcClientConfiguration.kt
package com.example.springgrpc.app

import com.example.springgrpc.client.HelloGrpcClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class GrpcClientConfiguration {
    @Bean(destroyMethod = "close")
    fun helloGrpcClient(properties: GrpcTargetProperties): HelloGrpcClient =
        HelloGrpcClient.connect(
            host = properties.host,
            port = properties.port,
            timeoutSeconds = properties.timeoutSeconds,
        )
}
